apps/front-client/src/pages/courses/[id]/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { GetServerSidePropsContext } from 'next';
+import { getServerSession } from 'next-auth';
+import { NextRouter, useRouter } from 'next/router';
+import { ReactElement } from 'react';
+import Course from '../../../features/courses/components/Organisms/CourseDetail';
+import DashboardLayout from '../../../layout/DashboardLayout';
+import { routes } from '../../../routes';
+import { authOptions } from '../../api/auth/[...nextauth]';
+
+const CourseDetail = () => {
+  const router: NextRouter = useRouter();
+
+  const { id } = router.query;
+
+  return <Course id={id as string} />;
```

Comment:
Pourquoi c'est fait ici et pas directement dans le composant `<Course />` ?

Et vaut mieux faire ça :
```suggestion
  const router: NextRouter = useRouter();

  const { id } = router.query as string;

  return <Course id={id} />;
```
Plus propre et surtout que bah si tu veux réutiliser `id` tu vas à chaque fois remettre `as string`

Reply:
Pareil !!

Reply:
Conversion of type 'ParsedUrlQuery' to type 'string' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first. Je le laisse tel-quel.